lms/static/scripts/frontend_apps/components/EmailNotificationsPreferences.tsx

```diff
@@ -0,0 +1,96 @@
+import type { PanelProps } from '@hypothesis/frontend-shared';
+import { Button, Checkbox, Panel } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import type { StateUpdater } from 'preact/hooks';
+import { useCallback } from 'preact/hooks';
+
+import type { EmailNotificationsPreferences, WeekDay } from '../config';
+
+const dayNames: [WeekDay, string][] = [
+  [7, 'Sunday'],
+  [1, 'Monday'],
+  [2, 'Tuesday'],
+  [3, 'Wednesday'],
+  [4, 'Thursday'],
+  [5, 'Friday'],
+  [6, 'Saturday'],
+];
+
+export type EmailNotificationsPreferencesProps = {
+  selectedDays: EmailNotificationsPreferences;
+  setSelectedDays: StateUpdater<EmailNotificationsPreferences>;
```

Comment:
Using `StateUpdater` here does component to the fact that state is stored using `useState` in the parent component. That's convenient at the moment, though you might find a need to change it later.